### ISSUE TITLE
Spike: VAOS conversion rate

### ISSUE THREAD

**Issue by lenaecb** [department-of-veterans-affairs/va.gov-team#8031]
Right now our conversion has been holding steady around 41- 45% respectively. 

We have the following hypotheses for this:
- **Hypothesis 1:** We believe that the majority of abandonment for request flows is due to: eligibility for request at their facility, or eligibility for request of their care type (not all facilities support a given type of care).
- **Hypothesis 2:** We believe the remaining abandonment is due to Veterans starting clicking “new appointment” button, but then promptly leave tool because they are just trying out the new interface.

_For the purpose of this Spike on conversion we are only going to focus on "VA requests" since CC and self-schedule is currently turned off._ 

**Questions we want to answer:**
- Are our hypotheses true?
   - If true - what % of users is this impacting type of care and/or facility isn't showing?
   - Can we get a list of the clinic IDs and care type pairs that are stopping users from proceeding?
   - Can we confirm that the clinic IDs and care types are not supported by specific facility by cross-checking VATS config /CDW? 
   - What % of users are coming into the flow and just leaving?
- Are there other underlying errors causing issues for Veterans who enter in to the scheduling flow?
- For submitted VA requests - why would a submitted request fail?
    - How often is that happening?
    - Why is that happening?

**AC:**
- Standardize our method for calculating conversions - user based vs. event based; submission vs. successful submissions
- Ticket for creating on-going, external facing reports of eligibility clinic ids based on errors we are encounter in VAOS - this report should be used to cross check VATS config per facility (Kay and team specifically interested, helps us understand % of conversions impacted by this)

**Supporting data/dashboards:**
- [VAOS GA Dashboard (v3)](https://analytics.google.com/analytics/web/?authuser=0#/dashboard/BtogOWE9Rp2oaAUz-EbwtQ/a50123418w177519031p184318975/_u.date00=20200324&_u.date01=20200414/)
- [Total GA Events](https://analytics.google.com/analytics/web/#/report/content-event-events/a50123418w177519031p176188361/_u.date00=20200324&_u.date01=20200414&explorer-segmentExplorer.segmentId=analytics.eventLabel&explorer-table.plotKeys=%5B%5D&explorer-table.filter=vaos&explorer-table.rowCount=25&explorer-table-tableMode.selected=data/)
- [Total Appointment Conversion funnel](https://analytics.google.com/analytics/web/?authuser=0#/my-reports/x7s5IXNxRFyRg79Oz6cTAg/a50123418w177519031p184318975/_u.date00=20200324&_u.date01=20200414/)
- [VA Appointment Request Conversion funnel](https://analytics.google.com/analytics/web/?authuser=0#/my-reports/BaEnCJBSR6G618aw0JQE1w/a50123418w177519031p184318975/_u.date00=20200324&_u.date01=20200414/)

**Comment by lenaecb** [comment]
@joanneesteban @jonwehausen - Want to tag you on this for awareness. We are going to spend some cycles this sprint understanding this data and why our users are dropping off. I know that we are actively working to improve our conversion data and understanding. I have added some points in this ticket about standardized how we look at conversions for VAOS-R.

**Comment by lenaecb** [comment]
@jonwehausen - The team and I met today to talk about ways we could dissect this. What we came up with though requires your/VSP Analytics team help. 

Here's a recap of our thoughts:
- We believe high number of abandonment is happening on the facilities page
     - How many people are triggering one of the eligibility events on this page?
     - From our analytics - Most common failed check/eligibility event comes from this check - Facilities
- **Question for analytics team:** Potential to pass along clinic id with facilities eligibility event? (@jonwehausen)

**Options for segmenting users/data more granular:** ( @jonwehausen @joanneesteban - we would need your help. Understanding our conversions more is a goal for this sprint so would love to implement the right option here soon.) 
1. Segment users who abandon on the facilities page -- using the Custom Appointment Conversion funnel, add user segment
      - Of those that were abandoned, how many are receiving and eligibility check/event?
2. Who encountered an eligibility check/event on the facilities page?
     - Of those that encountered, who went on to get the “VA request path started”
     - Those that remain are falling out because of facilities event check constraint
3. Better understand what people are doing on the facilities page
      - Track clicks/selections on the page / counter
      - Are they going back to the ‘type of care page’ then starting again (can this be part of the funnel)
Action - review Sentry errors for systems and facilities 

**Additional Action items/ideas for the VAOS_R team:**
- Potential future need - research the facilities page for naming convention; hypothesis that the naming in new vaos is confusing for systems and facilities

**Comment by jonwehausen** [comment]
Thank you for this @lenaecb ! This is very helpful for us in determining how we can segment to best suit your needs. 

Give us some time to explore the options you laid out in how we may best deliver on validation/devalidating your hypotheses.

**Comment by jonwehausen** [comment]
Hi Lenae, 

Here's some of my initial findings: 
- [Take a look at this PDF...](https://github.com/department-of-veterans-affairs/va.gov-team/files/4512023/Analytics.Health.Care.Modernized.View.VAOS.-.Ineligibility.Flags.-.Exit.Pages.20200414-20200420.pdf). What you're seeing is the users who clicked "Start Appointment", received an ineligibility flag, abandoned the funnel and where they exited. 
- Over the last week (date range in PDF report), your hypothesis holds true that 1 out of every 
4 users are receiving ineligibility flags and exiting on the facilities page. 
- Unfortunately, we're not capturing the care type at the point they receive the ineligibility flag, so we would need to work with you all to make that config change to get a better understanding of what care type & facility pairs are causing dropoff.

**Comment by lenaecb** [comment]
Adding this note from Jeff's work on the Sentry error discovery: #8173
Errors on facility page in FE Sentry since 3/24:

Timeouts: 10
Bad gateway error: 19
Misc network/client errors: 3
Bad request error: 14 (all on 3/24 due to FE bug, none since)
I would say that then that we have 46 instances of someone being blocked on the facility page. I don't know if you can directly compare that to our conversion numbers, but it would cover 7% at most of the 626 users who didn't started the flow but didn't finish (and be responsible for a 3% drop in the conversion rate).

We've already addressed 4), and 3) is not something we can reasonably impact. 1) and 2) are things we can research and possibly address.

**Comment by lenaecb** [comment]
We have been able to answer most of these questions with GA custom funnels and our existing GA dashboard. 
- [Full funnel for VA Requests ](https://analytics.google.com/analytics/web/#/analysis/a50123418w177519031p176188361/edit/5pTuYti2SICJcXOmDUObBA)- shows various steps where user is dropping off (right now 33% dropoff on Facilities page; 17% on Type of Care; 11% on Choose date/time)
- Specific eligibility drop-off on Facilities page - [Funnel](https://analytics.google.com/analytics/web/?authuser=0#/my-reports/x7s5IXNxRFyRg79Oz6cTAg/a50123418w177519031p184318975/_u.date00=20200324&_u.date01=20200422/)
- Specific "facility not supported" breakdown of eligibility alerts - [funnel](https://analytics.google.com/analytics/web/?authuser=0#/my-reports/x7s5IXNxRFyRg79Oz6cTAg/a50123418w177519031p184318975/_u.date00=20200324&_u.date01=20200423/)

I have created a new ticket to start tracking facility id and care type when eligibility event is fired #8367
